iroh-relay/src/dns.rs

```diff
@@ -263,6 +280,82 @@ impl DnsResolver {
         let f = || self.lookup_node_by_id(node_id, origin);
         stagger_call(f, delays_ms).await
     }
+
+    /// Look up a [`RelayNode`] and return a [`SocketAddr`].
+    ///
+    /// Uses `[DnsResolver::lookup_ipv4_staggered]`.
+    ///
+    /// Used for establishing ipv4 QUIC connections to relay nodes.
+    pub async fn lookup_relay_node_ipv4(
+        &self,
+        relay: &RelayNode,
+        timeout: Duration,
+        delays_ms: &[u64],
+        port: u16,
+    ) -> Result<SocketAddr> {
```

Comment:
You mention these two points in the notes of the PR:
- For name I think "relay" doesn't say anything.  Maybe `lookup_node_ipv4_and_map` would be better.
- Definitely return an `IpMappedAddr` instead of a `SocketAddr`.